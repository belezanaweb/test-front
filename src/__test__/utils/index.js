import checkPropTypes from 'check-prop-types'

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`)
  return wrapper
}

export const checkProps = (component, expectedProps) => {
  const propsError = checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
  return propsError
}
