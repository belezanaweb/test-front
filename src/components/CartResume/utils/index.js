export const normalizeCartData = data => {
  return {
    productsQuantity: data.items.length,
    subTotal: data.subTotal.toFixed(2).replace('.', ','),
    shipping: data.shippingTotal.toFixed(2).replace('.', ','),
    discount: data.discount.toFixed(2).replace('.', ','),
    total: data.total.toFixed(2).replace('.', ','),
  };
};
