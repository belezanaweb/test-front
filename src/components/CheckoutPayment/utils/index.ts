import * as Yup from 'yup'

const errors = {
  name: 'Insira um nome válido',
  code: 'Insira um código válido',
  number: 'Insira um número válido',
  expiration: 'Insira uma data válida',
}

export const initialValues = {
  creditCardName: '',
  creditCardNumber: '',
  creditCardExpiration: '',
  creditCardCode: '',
}

const regexOnlyNumbers = /[^\d.-]+/g

export const CheckoutPaymentSchema = Yup.object().shape({
  creditCardName: Yup.string()
    .min(2, errors.name)
    .matches(/^([A-Za-z\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff\s]*)$/gi, errors.name)
    .required(errors.name),
  creditCardNumber: Yup.string()
    .min(2, errors.number)
    .test('len', errors.number, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 16
    })
    .required(errors.number),
  creditCardExpiration: Yup.string()
    .min(2, errors.name)
    .test('len', errors.expiration, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 4
    })
    .required(errors.expiration),
  creditCardCode: Yup.string()
    .matches(/^[0-9]+$/, errors.code)
    .test('len', errors.code, (val) => val.length === 3)
    .required(errors.code),
})
