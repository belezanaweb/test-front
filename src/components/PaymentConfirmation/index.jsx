import React from 'react'
import iconImg from '../../assets/icon/confirmation.svg'
import * as S from './styled'

const PaymentConfirmation = ({ status }) => {
  const successPaymentInfo = {
    success: {
      text: 'Compra efetuada com sucesso',
      color: '#FF7800',
      icon: iconImg
    }
  }

  const { icon, text, color } = successPaymentInfo[status]

  return (
    <S.PaymentConfirmationContainer>
      <S.PaymentConfirmationContainerImage>
        <S.PaymentConfirmationImage src={icon} />
      </S.PaymentConfirmationContainerImage>
      <S.PaymentConfirmationText color={color}>{text}</S.PaymentConfirmationText>
    </S.PaymentConfirmationContainer>
  )
}

export default PaymentConfirmation
