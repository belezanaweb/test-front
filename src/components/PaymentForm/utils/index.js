export const formatCardNumber = value => {
  return value
    .replace(/\D/g, '')
    .replace(/(\d{4})(?=\d)/g, '$1 ')
    .substr(0, 19);
};

export const formatHolderName = value => {
  return value.replace(/[^a-zA-Z\u00C0-\u017F\s]/gu, '').substr(0, 60);
};

export const formatExpirationDate = value => {
  return value
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d{0,4})/, '$1/$2')
    .substr(0, 7);
};

export const formatCVVCode = value => {
  return value.replace(/\D/g, '').substr(0, 3);
};

const validateCreditCard = cardNumber => {
  cardNumber = cardNumber.replace(/\s/g, '');

  if (cardNumber.length !== 16) {
    return false;
  }

  const digits = cardNumber.split('').map(Number);

  // Algoritmo de Luhn para validar o número do cartão
  let sum = 0;

  for (let i = 0; i < digits.length; i++) {
    let digit = digits[i];

    if (i % 2 === 0) {
      digit *= 2;

      if (digit > 9) {
        digit -= 9;
      }
    }

    sum += digit;
  }

  // O número do cartão é válido se a soma dos dígitos for um múltiplo de 10
  return sum % 10 === 0;
};

const isInvalidDate = dateString => {
  const today = new Date();
  const [month, year] = dateString.split('/');
  if (month === '00') {
    return true;
  }
  const userDate = new Date(Number(year), Number(month) - 1);

  return userDate < new Date(today.getFullYear(), today.getMonth());
};

export const validateForm = state => {
  const errors = {};

  if (!validateCreditCard(state.cardNumber)) {
    errors.cardNumber = 'insira um número de cartão válido';
  }

  if (!state.cardHolderName) {
    errors.cardHolderName = 'insira um nome válido';
  }

  if (state.expirationDate.length < 7 || isInvalidDate(state.expirationDate)) {
    errors.expirationDate = 'insira uma data válida';
  }

  if (state.cvv.length < 3) {
    errors.cvv = 'insira um cvv válido';
  }

  return errors;
};
