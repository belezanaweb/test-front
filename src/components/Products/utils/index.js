export const normalizeCartData = data => {
  return data.items.reduce((acc, curr) => {
    const formattedProduct = {
      imageUrl: curr.product.imageObjects[0].small,
      productName: curr.product.name,
      price: curr.product.priceSpecification.price,
      maxPrice: curr.product.priceSpecification.maxPrice,
    };

    return [...acc, formattedProduct];
  }, []);
};
