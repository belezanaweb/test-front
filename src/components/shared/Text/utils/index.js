export { colors, decorations, sizes, weights } from './cssTokens.js';
