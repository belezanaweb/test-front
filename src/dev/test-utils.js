// source: https://testing-library.com/docs/react-testing-library/setup#custom-render
import React, { Suspense } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import theme from '../theme'

// eslint-disable-next-line react/prop-types
function WithProviders ({ children }) {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback={<>Loading...</>}>{children}</Suspense>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: WithProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
