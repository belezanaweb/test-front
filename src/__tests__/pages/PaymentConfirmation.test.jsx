import React from 'react'
import { render } from '../../utils/test-utils'
import PaymentConfirmation from '../../pages/PaymentConfirmation'

jest.mock('../../hooks/checkout.jsx', () => ({
  useCheckout: jest.fn().mockReturnValue({
    cartItems: [],
    cartSummary: {},
    isLoading: false,
    paymentInfo: {
      cardName: 'sample',
      cardNumber: '1234.1234.1234.1234',
      cardValid: '07/1991',
      cardCode: '1234'
    },
    submitPayment: jest.fn()
  })
}))

describe('PaymentConfirmation Page', () => {
  test('should render without errors', () => {
    const { container } = render(<PaymentConfirmation />)
    expect(container).toBeInTheDocument()
  })

  test('should match with snapshot', () => {
    const { container } = render(<PaymentConfirmation />)
    expect(container).toMatchSnapshot()
  })
})
