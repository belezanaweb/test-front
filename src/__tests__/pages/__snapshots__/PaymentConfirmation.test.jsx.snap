// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentConfirmation Page should match with snapshot 1`] = `
<div>
  <main
    class="sc-eCApnc iHOios"
  >
    <header
      class="sc-bdnxRM cZjwSm"
    >
      <div
        class="sc-gtsrHT egWxvy"
      >
        <nav
          class="sc-dlnjwi kDoCbM"
        >
          <a
            class="sc-hKFxyN vGRFp"
            href="/cart"
          >
            Sacola
          </a>
          <a
            class="sc-hKFxyN vGRFp"
            href="/payment"
          >
            Pagamento
          </a>
          <a
            class="sc-hKFxyN vGRFp"
            href="/confirmation"
          >
            Confirmação
          </a>
        </nav>
      </div>
    </header>
    <div
      class="sc-jSFjdj drOsWB"
    >
      <div
        class="sc-dlMDgC hyUoTp"
      >
        <svg
          class="sc-fKgJPI fKlqQ"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
          />
        </svg>
        <p
          class="sc-kfYoZR fRurXo"
        >
          Compra efetuada com sucesso
        </p>
      </div>
      <section
        class="sc-iCoGMd"
      >
        <h2
          class="sc-fujyAs djPjrY"
        >
          Pagamento
        </h2>
        <div
          class="sc-pNWdM fRwEYN"
        >
          <div
            class="sc-bYwzuL irqKRJ"
          >
            <p
              class="sc-kLojOw bVLfhE"
            >
              ****.****.****.1234
            </p>
            <p
              class="sc-kLojOw bVLfhE"
            >
              sample
            </p>
            <p
              class="sc-kLojOw bVLfhE"
            >
              07/1991
            </p>
          </div>
        </div>
      </section>
      <section
        class="sc-iCoGMd"
      >
        <h2
          class="sc-fujyAs djPjrY"
        >
          Produtos
        </h2>
        <div
          class="sc-pNWdM fRwEYN"
        >
          <div
            class="sc-jrsJWt dqothG"
          >
            <ul />
          </div>
        </div>
      </section>
      <section
        class="sc-ksluID iNvcqR"
      >
        <div
          class="sc-fFSPTT sc-bkbkJK jPYEnV"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Produtos
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
        <div
          class="sc-fFSPTT sc-iemWCZ jPYEnV"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Frete
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
        <div
          class="sc-fFSPTT sc-dIvrsQ jPYEnV hRyHfU"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Desconto
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
        <div
          class="sc-fFSPTT sc-hHEiqL jPYEnV cFkeXV"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Total
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
      </section>
    </div>
  </main>
</div>
`;
